code/modules/research/stockparts.dm

```diff
@@ -1,11 +1,86 @@
+#define STOCK_PART_BASIC 1
+#define STOCK_PART_ADVANCED 2
+#define STOCK_PART_SUPER 3
+
+/obj/item/storage/bag/stockparts_box
+	name = "stock parts box"
+	desc = "A low-tech method of storing stock parts used in machinery."
+	icon = 'icons/obj/stock_parts.dmi'
+	icon_state = "empty_box"
+	w_class = ITEMSIZE_NORMAL
+	display_contents_with_number = TRUE
+	storage_slots = 25
+	can_hold = list(
+				/obj/item/stock_parts,
+				/obj/item/device/multitool,
+				/obj/item/stack/cable_coil
+					)
+	max_w_class = ITEMSIZE_SMALL
+	max_storage_space = 60
+	matter = list(DEFAULT_WALL_MATERIAL = 750)
+	var/list/spawn_items = list()
```

Comment:
Storage items have a `starts_with` list for this.

Reply:
mfw